### ISSUE TITLE
improved support for multiple clients

### ISSUE THREAD

**Issue by brainchild0** [borgbackup/borg#7090]
Regarding the issue of using the same repository from multiple clients, the FAQ notes the following two limitations:

1. The data deposited to the repository is not secure against tampering, normally protected by the client's cryptographic key.
2. General use  of the application depends on data cached on the client.

Both cache and cryptography are useful features for certain cases, but can be limiting.

In many cases, various systems maintain   data sets that vary only slightly compared to the overall size. The de-duplication mechanism of a backup system would offers both a compact means for  storing the various,    different data sets, as well as for efficiently identifying the differences.

Would stronger support for such an expanded range of use cases be considered for future development?

**Comment by ThomasWaldmann** [comment]
1. that is due to aes-ctr usage (the one client can not know how much the other client has increased the counter and thus needs to rely on the repo). this will be solved by borg2 by not using aes-ctr any more / by using session keys.
2. you can also use borg without the cache, it will be automatically rebuilt.

so, guess that solves this issue?

**Comment by brainchild0** [comment]
> so, guess that solves this issue?

The answer largely depends the choice of whether the scope of the issue would include the FAQ and other documentation, or only the software base.

Regarding cache, it seems, there is a case for its being disabled completely in certain instances. It consumes local storage, and the data probably may be easily regenerated on modern hardware.

**Comment by ThomasWaldmann** [comment]
The biggest space consumption is by chunks.archives.d and there is already a FAQ entry about that.

If there's anything else, please be more specific.

**Comment by brainchild0** [comment]
I have made a variety of comments in past weeks, of varying levels of specificity, that seemed appropriate to the various issues.

At the moment, the issue seeming most relevant to me is quite general, about getting a sense of the range of readily accessible use cases being much more narrowly constrained than is necessary or helpful to many users.

In the past, I invoked a comparison to Git,  though not to suggest that it is an appropriate tool for backups. Instead, I think it is useful to note that despite the steep learning curve  for many features of the tool, the basic operations of managing repositories are clear. It is clear how to initialize a new repository, to commit new data to it, to extract data from it,  to move data between repositories, and to move repositories to new locations in the same system (the last being a simple file operation). 

Taking a case of a tool more appropriate to backups, the Zpac archival utility has many similar features as Borg, though the incremental feature   is based simply on appending to existing archives, without  support for removal of previous commits   and for client/server operation. Nevertheless, it is instructive to consider the clarity of basic operations. To *add* command simply adds a file set to the archive. It partitions the file set into chunks, computes the hash of each chunk, and writes them to the target file. If the target exists, it compares the set of computed hashes to the set of existing ones, and writes only the new chunks. If the *force* option is not specified, it skips any files with a  modification time more recent than that of the archive. It is very accessible and intuitive.

Encryption and cache are useful features, but also potential sources for undesirable complications. Each tool I mentioned has considerable similarity to Borg across various sets of features, and allows a basic  invocation that is not burdened by features   not universally necessary. Neither tool is perfect, of course. It is for instance an extremely considerable vulnerability that Zpac only accepts passwords passed directly on the command line.

At the present moment, getting started with Borg, even for basic usage, seems to me much the same as navigating a maze of narrow passages and high walls. I have no opinion on whether the effect is more substantially from the application design itself versus the documentation, but such is my feeling, all told.
